integrations/opensearch/src/haystack_integrations/common/opensearch/utils.py

```diff
@@ -0,0 +1,53 @@
+from typing import Optional
+
+from haystack.lazy_imports import LazyImport
+from haystack_integrations.common.opensearch.errors import AWSConfigurationError
+
+with LazyImport("Run 'pip install \"boto3\"' to install boto3.") as boto3_import:
+    import boto3
+    from botocore.exceptions import BotoCoreError
+
+AWS_CONFIGURATION_KEYS = [
+    "aws_access_key_id",
+    "aws_secret_access_key",
+    "aws_session_token",
+    "aws_region_name",
+    "aws_profile_name",
+]
+
+
+def get_aws_session(
```

Comment:
I would move this in `auth.py` and make it private. It's used only there in any case.